src/camera-helper.ts

```diff
@@ -0,0 +1,143 @@
+/*
+  Copyright 2017 Google Inc. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import constants from './constants';
+import {canvasToBlob} from './promise-helpers';
+
+const streamConstraints: MediaStreamConstraints = {
+  audio: false,
+  video: {
+    deviceId: '',
+    facingMode: ['user', 'environment'],
+    height: {ideal: 1080},
+    width: {ideal: 1920},
+  },
+};
+
+const supportedConstraints = navigator.mediaDevices.getSupportedConstraints();
```

Comment:
I think this is gonna fail in some browsers due to `SUPPORTS_MEDIA_DEVICES` being `false`.

Reply:
Oops, good catch